tetherback/tetherback.py

```diff
@@ -124,10 +124,18 @@ def sensible_transport(transport, adbversion):
             return adbxp.tcp
     return transport
 
+def get_userdata_partinfo(fstab):
+    for dev, info in fstab.items():
+        if info[0] == '/data':
+            return { 'dev': dev.split('/')[-1], 'mnt': info[0], 'fs': info[1] }
```

Comment:
Well, it's the complete absolute path, e.g. /dev/block/dm-0 but the other entries in your partition map contain just the last component. I had to be consistent otherwise I would break the subsequent functions. :)